interfaces/cython/cantera/examples/onedim/ConvoluteCpp.h

```diff
@@ -0,0 +1,187 @@
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Copyright (C) 2017 The University Of Dayton. All Rights Reserved.
+//
+// No part of this program may be photocopied, transferred, or otherwise reproduced in machine or
+// human readable form without the prior written consent of The University Of Dayton.
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESSED OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+// FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// UNIVERSITY OF DAYTON OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+// OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+// EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.� THE UNIVERSITY OF DAYTON
+// HAS NO OBLIGATION TO SUPPORT THE SOFTWARE.
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <vector>
+#include <stdio.h>
+
+//! Vector of doubles.
+typedef std::vector<double> vector_fp;
+//! Vector of ints
+typedef std::vector<int> vector_int;
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/// 
+/// <version>1.0.0</version>
+/// <author>Robert B. Olding</author>
+///
+/// <summary>

```

Comment:
Documentation should be provided in the format of Doxygen docstrings. Please see the other C++/.h files for examples